pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -873,10 +874,11 @@ func (c *Cluster) startMon(m *monConfig, node *NodeInfo) error {
 		if c.Network.IsHost() || !pvcExists {
 			p.PodAffinity = nil
 			p.PodAntiAffinity = nil
-			c.setPodPlacement(&d.Spec.Template.Spec, p,
-				existingDeployment.Spec.Template.Spec.NodeSelector)
+			k8sutil.SetNodeAntiAffinityForPod(&d.Spec.Template.Spec, p, requiredDuringScheduling(c), true,
```

Comment:
Please use a var instead of true, it's hard to understand what `true` means without looking at the definition.